plugins/http_plugin/http_plugin.cpp

```diff
@@ -287,13 +328,202 @@ namespace eosio {
+         }
+
+         /**
+          * Helper type that wraps an object of type T and records its "in flight" size to
+          * http_plugin_impl::bytes_in_flight using RAII semantics
+          *
+          * @tparam T - the contained Type
+          */
+         template<typename T>
+         struct in_flight {
+            in_flight(T&& object, http_plugin_impl& impl)
+            :object(std::move(object))
+            ,impl(impl)
+            {
+               count = detail::in_flight_sizeof(object);
+               impl.bytes_in_flight += count;
+            }
+
+            ~in_flight() {
+               if (count) {
+                  impl.bytes_in_flight -= count;
+               }
+            }
+
+            // No copy constructor, but allow move
+            in_flight(const in_flight&) = delete;
+            in_flight(in_flight&& from)
+            :object(std::move(from.object))
+            ,count(from.count)
+            ,impl(from.impl)
+            {
+               from.count = 0;
+            }
+
+            // No copy assignment, but allow move
+            in_flight& operator=(const in_flight&) = delete;
+            in_flight& operator=(in_flight&& from) {
+               object = std::move(from.object);
+               count = from.count;
+               impl = from.impl;
+               from.count = 0;
+            }
+
+            /**
+             * const accessor
+             * @return const reference to the contained object
+             */
+            const T& operator* () const {
+               return object;
+            }
+
+            /**
+             * mutable accessor (can be moved frmo)
+             * @return mutable reference to the contained object
+             */
+            T& operator* () {
+               return object;
+            }
+
+            T object;
+            size_t count;
+            http_plugin_impl& impl;
+         };
+
+         /**
+          * convenient wrapper to make an in_flight<T>
+          */
+         template<typename T>
+         static auto make_in_flight(T&& object, http_plugin_impl& impl) {
+            return in_flight<T>(std::forward<T>(object), impl);
+         }
+
+         /**
+          * Make an internal_url_handler that will run the url_handler on the app() thread and then
+          * return to the http thread pool for response processing
+          *
+          * @pre b.size() has been added to bytes_in_flight by caller
+          * @param priority - priority to post to the app thread at
+          * @param next - the next handler for responses
+          * @return the constructed internal_url_handler
+          */
+         detail::internal_url_handler make_app_thread_url_handler( int priority, url_handler next ) {
+            auto next_ptr = std::make_shared<url_handler>(std::move(next));
+            return [this, priority, next_ptr]( detail::abstract_conn_ptr conn, string r, string b, url_response_callback then ) mutable {
```

Comment:
Minor thing: you could `std::move(next_ptr)` here, but this is only called once on startup so no big deal.